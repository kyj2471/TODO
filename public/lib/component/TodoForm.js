'use strict';Object.defineProperty(exports,'__esModule',{value:!0});var _slicedToArray=function(){function a(a,b){var c=[],d=!0,e=!1,f=void 0;try{for(var g,h=a[Symbol.iterator]();!(d=(g=h.next()).done)&&(c.push(g.value),!(b&&c.length===b));d=!0);}catch(a){e=!0,f=a}finally{try{!d&&h['return']&&h['return']()}finally{if(e)throw f}}return c}return function(b,c){if(Array.isArray(b))return b;if(Symbol.iterator in Object(b))return a(b,c);throw new TypeError('Invalid attempt to destructure non-iterable instance')}}(),_react=require('react'),_react2=_interopRequireDefault(_react),_reactRedux=require('react-redux'),_todo=require('../store/modules/todo'),_styles=require('../../styles/styles'),S=_interopRequireWildcard(_styles);function _interopRequireWildcard(a){if(a&&a.__esModule)return a;var b={};if(null!=a)for(var c in a)Object.prototype.hasOwnProperty.call(a,c)&&(b[c]=a[c]);return b.default=a,b}function _interopRequireDefault(a){return a&&a.__esModule?a:{default:a}}function TodoForm(a){var b=a.addTodoSaga,c=(0,_react.useState)(''),d=_slicedToArray(c,2),e=d[0],f=d[1],g=function(){b({id:Date.now(),text:e}),f('')};return<S.HeadTodoForm>
      <form className="formContainer"onSubmit={g}>
        <S.TodoInput className="todoInput"value={e}onChange={function b(a){a.preventDefault(),f(a.target.value)}}type="text"required/>
        <S.MainButton huge type="submit"className="todoButton"onClick={g}>
          추가
        </S.MainButton>
      </form>
    </S.HeadTodoForm>}var mapStateToProps=function(a){var b=a.text,c=a.addTodoSaga;return{text:b,addTodoSaga:c}},mapDispatchToProps=function(a){return{addTodoSaga:function c(b){return a((0,_todo.addTodoSaga)(b))}}};exports.default=(0,_reactRedux.connect)(mapStateToProps,mapDispatchToProps)(TodoForm);