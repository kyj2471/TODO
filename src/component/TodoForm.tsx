import React, { useState } from 'react';
import { connect } from 'react-redux';
import { addTodoSaga } from '../store/modules/todo';
import * as S from '../../styles/styles';

interface TodoForm {
  text: any;
  addTodoSaga: any;
}

function TodoForm({ addTodoSaga }: TodoForm) {
  const [input, setInput] = useState('');

  const handleChange = (e: any) => {
    e.preventDefault();
    setInput(e.target.value);
  };

  const handleSubmit = () => {
    addTodoSaga({
      id: Date.now(),
      text: input
    });

    setInput('');
  };

  return (
    <S.HeadTodoForm>
      <form className="formContainer" onSubmit={handleSubmit}>
        <S.TodoInput
          className="todoInput"
          value={input}
          onChange={handleChange}
          type="text"
          required
        />
        <S.MainButton
          huge
          type="submit"
          className="todoButton"
          onClick={handleSubmit}>
          추가
        </S.MainButton>
      </form>
    </S.HeadTodoForm>
  );
}

const mapStateToProps = ({ text, addTodoSaga }: TodoForm) => ({
  text,
  addTodoSaga
});

const mapDispatchToProps = (dispatch: any) => ({
  addTodoSaga: (todo: TodoForm) => dispatch(addTodoSaga(todo))
});

export default connect(mapStateToProps, mapDispatchToProps)(TodoForm);
