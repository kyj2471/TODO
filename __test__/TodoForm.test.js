import React from 'react';
import TodoForm from '../src/component/TodoForm';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';

const mockStore = configureStore([]);
let store;
let component;

beforeEach(() => {
  store = mockStore({
    myState: 'sample text'
  });
  store.dispatch = jest.fn();

  component = renderer.create(
    <Provider store={store}>
      <TodoForm />
    </Provider>
  );
});

describe('<T O D O _ F O R M > T E S T', () => {
  it('[스냅샷]-TodoForm', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('todoform test', () => {
    jest.spyOn(Date, 'now').mockReturnValueOnce(1000);
    renderer.act(() => {
      component.root.findByType('button').props.onClick();
    });

    renderer.act(() => {
      component.root
        .findByType('input')
        .props.onChange({ target: { value: 'asd' } });
    });

    expect(store.dispatch).toHaveBeenCalledTimes(1);
    expect(store.dispatch).toHaveBeenCalledWith({
      payload: {
        id: 1000,
        text: ''
      },
      type: 'todo/ADD_TODO_SAGA'
    });
  });
});

// import React from 'react';
// import { render, fireEvent, screen } from '@testing-library/react';
// import TodoForm from '../src/component/TodoForm';
// import configureStore from 'redux-mock-store';

// const mockStore = configureStore([]);

// let store = mockStore({
//   myState: 'sample text'
// });
// store.dispatch = jest.fn();
// console.log(store);

// it('should not call addTodo if length of text is 0', () => {
//   const mockAddTodoSaga = jest.fn();
//   render(<TodoForm store={store} addTodoSaga={mockAddTodoSaga} />);
//   fireEvent.change(screen.getByLabelText(/formInput/i), {
//     target: { value: '' }
//   });
//   expect(mockAddTodoSaga).toHaveBeenCalledTimes(0);
// });

// it('should call addTodo if length of text is greater than 0', () => {
//   const mockAddTodoSaga = jest.fn();
//   render(<TodoForm store={store} addTodoSaga={mockAddTodoSaga} />);

//   fireEvent.change(screen.getByLabelText(/formInput/i), {
//     target: { value: 'Tony' }
//   });
//   expect(mockAddTodoSaga).toHaveBeenCalledTimes(1);
// });
