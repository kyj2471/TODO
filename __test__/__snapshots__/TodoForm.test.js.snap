// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<T O D O _ F O R M > T E S T [스냅샷]-TodoForm 1`] = `
<div
  className="sc-dkPtyc eBcJGs"
>
  <form
    className="formContainer"
    onSubmit={[Function]}
  >
    <div>
      asdf
    </div>
    <input
      aria-label="help"
      className="sc-hKwCoD gBceBh todoInput"
      onChange={[Function]}
      placeholder="hi"
      required={true}
      type="text"
      value=""
    />
    <button
      aria-label="Enter2"
      className="sc-bdvvaa iRUeFJ todoButton"
      onClick={[Function]}
      type="submit"
    >
      추가
    </button>
  </form>
</div>
`;
